### ISSUE TITLE
Cannot DROP and reCREATE extension

### ISSUE THREAD

**Issue by lithp** [citusdata/citus#1233]
```
brian=# CREATE EXTENSION citus;
CREATE EXTENSION
brian=# DROP EXTENSION citus;
DROP EXTENSION
brian=# CREATE EXTENSION citus;
ERROR:  could not open relation with OID 16395
CONTEXT:  SQL statement "SELECT EXISTS(SELECT * FROM pg_proc WHERE proname = 'worker_apply_shard_ddl_command')"
PL/pgSQL function inline_code_block line 3 at IF
```

**Comment by marcocitus** [comment]
The relation that could not be opened appears to be pg_dist_partition.

**Comment by marcocitus** [comment]
Also, all other queries also fail with the same error.

**Comment by marcocitus** [comment]
Ok, so we never register the invalidation handler if you drop extension immediately after creating, since we never call IsDistributedTable. However, we do call CitusHasBeenLoaded which sets distShardRelationId to the OID of pg_dist_partition, which then gets dropped. Any subsequent call to IsDistributedTable (which in Brian's example happens in the `SELECT EXISTS ...`) would then try to use the stale OID and fail.

**Comment by anarazel** [comment]
So we need to have CachedRelationLookup() setup all the invalidation handlers.  I suspect we should centralize the setup of that into one function (from InitializeDistTableCache(), InitializeWorkerNodeCache(), GetLocalGroupId()) and then make sure that it's being called in all the necessary places.

**Comment by marcocitus** [comment]
It might make sense to set up invalidation handlers from CitusHasBeenLoaded, since it caches something prior to setting up invalidation handlers.

**Comment by jasonmp85** [comment]
This is something worth backporting, right?

**Comment by anarazel** [comment]
> This is something worth backporting, right?

Yes.

**Comment by metdos** [comment]
@byucesoy, I couldn't replicate this problem with your updated branch on this PR #1427.

PS. This is not a blocker for 6.2 release.

**Comment by marcocitus** [comment]
Fixed by #1461
